sdks/go/test/integration/primitives/checkpointing_test.go

```diff
@@ -18,15 +18,11 @@ package primitives
 import (
 	"testing"
 
-	"github.com/apache/beam/sdks/v2/go/pkg/beam"
 	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
 	"github.com/apache/beam/sdks/v2/go/test/integration"
 )
 
 func TestCheckpointing(t *testing.T) {
 	integration.CheckFilters(t)
-
-	p, s := beam.NewPipelineWithRoot()
-	Checkpoints(s)
-	ptest.RunAndValidate(t, p)
+	ptest.BuildAndRun(t, Checkpoints)
```

Comment:
I really like the cleanliness of this new pattern

Reply:
I've been leaning this way for a long time too, and it will mesh nicely with a proper options per-pipeline specifier, rather than *everything is flags*.